maths/prime_number.rb

```diff
@@ -0,0 +1,38 @@
+# A ruby program to check a given number is prime or not
+# Mathematical explanation: A number which has only 2 factors i.e., 1 (one) and itself
+
+# Prime number check function
+def is_prime_number(number)
+  primeFlag = false
+  if number>0
+    if number == 1
+    primeFlag = true
+    else
+      factors = 2;
+      while factors.to_i <= number.to_i/2
+        if number.to_i % factors.to_i==0
+        primeFlag = true
+        break
+        end
+        factors+=1
+      end
+    end
+  else
+  primeFlag = true
+  end
+
+  if !primeFlag
+    print "\nThe given number " + number.to_s + " is a Prime."
```

Comment:
String interpolation will make your code more concise. It calls `.to_s` automatically.


```diff
-    print "\nThe given number " + number.to_s + " is a Prime."
+    print "The given number #{number} is a Prime."
```

>Double-quote strings allow interpolation of other values using `#{...}`:
>
>```rb
>"One plus one is two: #{1 + 1}"
>```
>
>https://docs.ruby-lang.org/en/3.0.0/doc/syntax/literals_rdoc.html#label-Strings